L1Trigger/L1CaloTrigger/plugins/L1CaloJetProducer.cc

```diff
@@ -0,0 +1,1371 @@
+// -*- C++ -*-
+//
+// Package: L1CaloTrigger
+// Class: L1CaloJetProducer
+//
+/**\class L1CaloJetProducer L1CaloJetProducer.cc
+
+Description: 
+Beginning with HCAL TPs, create HCAL jet, then
+take L1EG crystal clusters from L1EGammaCrystalsProducer.cc
+and clusters them within fixed number of trigger towers
+
+Implementation:
+[Notes on implementation]
+*/
+//
+// Original Author: Tyler Ruggles
+// Created: Tue Aug 29 2018
+// $Id$
+//
+//
+
+#include "DataFormats/Math/interface/deltaR.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+
+#include <iostream>
+
+#include "DataFormats/L1Trigger/interface/L1JetParticleFwd.h"
+//#include "DataFormats/L1TCalorimeterPhase2/interface/CaloCrystalCluster.h"
+#include "DataFormats/L1TCalorimeterPhase2/interface/CaloJet.h"
+#include "DataFormats/L1TCalorimeterPhase2/interface/CaloTower.h"
+#include "DataFormats/L1THGCal/interface/HGCalTower.h"
+#include "DataFormats/HcalDigi/interface/HcalDigiCollections.h"
+
+#include "CalibFormats/CaloTPG/interface/CaloTPGTranscoder.h"
+#include "CalibFormats/CaloTPG/interface/CaloTPGRecord.h"
+
+#include "L1Trigger/L1TCalorimeter/interface/CaloTools.h"
+
+// For pT calibrations
+#include "TF1.h"
+
+// Run2/PhaseI output formats
+#include "DataFormats/L1Trigger/interface/Tau.h"
+#include "DataFormats/L1Trigger/interface/Jet.h"
+
+class L1CaloJetProducer : public edm::EDProducer {
+public:
+  explicit L1CaloJetProducer(const edm::ParameterSet &);
+
+private:
+  void produce(edm::Event &, const edm::EventSetup &) override;
+  //bool cluster_passes_base_cuts(float &cluster_pt, float &cluster_eta, float &iso, float &e2x5, float &e5x5) const;
+  int ecalXtal_diPhi(int &iPhi_1, int &iPhi_2) const;
```

Comment:
Where is this `ecalXtal_diPhi` used? 
Actually, it is almost the same as tower_diPhi, should you try to simplify them?

Reply:
tower_diPhi seems to be using units of tower positions while ecalXtal_diPhi seems to be using the phi position itself. Unfortunately I am not sure where ecalXtal_diPhi is used further down the line.